src/odemis/gui/cont/tabs.py

```diff
@@ -1961,12 +1961,15 @@ def __init__(self, name, button, panel, main_frame, main_data):
         self.view_controller = viewcont.ViewPortController(self.tab_data, panel, vpv)
 
         # Single-beam SEM stream
-        hwemt_vanames = ("resolution", "scale", "horizontalFoV")
-        emt_vanames = ("dwellTime")
-        hwdet_vanames = ("brightness", "contrast")
+        hwemt_vanames = ["horizontalFoV", "scale", "resolution"]
+        hwdet_vanames = ["contrast", "brightness"]
+        emt_vanames = ["dwellTime"]
+        # VA names as per the display order
+        # Process order is first hardware VAs, then emitter VAs and finally detector VAs
+        hw_settings_display_order = hwemt_vanames + hwdet_vanames + emt_vanames
```

Comment:
I'm not convinced by this approach. There is already a mechanism to sort the VAs. `get_hw_config()` returns the names in a order specified in HW_SETTINGS_CONFIG. Right now, the e-beam order is set to: horizontalFoV, dwellTime, scale, resolution. You want to change to horizontalFoV, scale, resolution, dwellTime. Actually, I don't think the current order is really logical. The other microscopes would also benefit from having dwellTime away, though I feel that it should be at the top. You probably could move dwellTime *up* in the HW_SETTINGS_CONFIG. As in the case of the FASTEM it's the only one which is a "local" VA, it will be at the bottom any way

The order for the `se-detector` is just not set at all. It would be easy to add an entry with contrast and brightness. You'd just need to indicate that this order is nice because it's the same as in the TFS software (so that no one  modifies it lightly). 

So, conclusion is that probably you can drop all the changes with `hw_settings_display_order`, and just tweak HW_SETTINGS_CONFIG to obtain the right thing. Can you try?

Reply:
I tried adding the order for `se-detector` in `HW_SETTINGS_CONFIG` and it works